Add 102 tutorial for CID generation configurations and CID profiles (#455)
ipfs-examples/helia-examples: examples/helia-101/README.md

Next we will look at where the bytes that make up the file go.

### 102 - CID Profiles

> The way UnixFS takes arbitrary inputs (files, folders, etc) and turns them into DAGs of smaller binary blocks (addressed by CID) involves a lot of tradeoffs and choices, particularly at the DAG level.

Reviewer: ```suggestion
The way UnixFS takes arbitrary inputs (files, folders, etc) and turns them into [DAGs](https://proto.school/merkle-dags/03) of smaller binary blocks (addressed by CID) involves a lot of tradeoffs and choices, particularly at the DAG level.
```

> If you need multiple different implementations to produce the same CID for a given input, you need to configure the CID generation accordingly, matching certain configuration variables to those defined in "CID profiles", like those exposed in the `ipfs config profile apply` command in kubo ([docs](https://docs.ipfs.tech/reference/kubo/cli/#ipfs-config-profile-apply)).

Reviewer: ```suggestion
If you need multiple different implementations to produce the same CID for a given input, configure CID generation accordingly, matching certain configuration variables to those defined in CID profiles, like those exposed in the `ipfs config profile apply` command in kubo ([docs](https://docs.ipfs.tech/reference/kubo/cli/#ipfs-config-profile-apply)).
```

> When experimenting with these options, it can be helpful to play with the DAG generator at dag.ipfs.tech.

Reviewer: ```suggestion
When experimenting with these options, it can be helpful to play with the [DAG Build](https://dag.ipfs.tech/) .
```
